myuw/dao/mailman.py

```diff
@@ -110,21 +114,46 @@ def get_section_email_lists(section,
         "course_number": section.course_number,
         "section_id": section.section_id,
         "is_primary": is_primary_section,
+        "has_multiple_sections": False,
+        "total_course_wo_list": 0,
         }
     json_data["section_list"] = get_single_section_list(section)
 
-    if is_primary_section and include_secondaries_in_primary:
+    if not json_data["section_list"]["list_exists"]:
+        json_data["total_course_wo_list"] = 1
+
+    if not is_primary_section:
+        json_data["no_secondary_section"] = True
+    else:
         total_secondaries = len(section.linked_section_urls)
+        json_data["no_secondary_section"] = (total_secondaries == 0)
+
         if total_secondaries > 0:
+            json_data["has_multiple_sections"] = True
+            if include_secondaries_in_primary:
+                json_data["secondary_section_lists"] =\
+                    get_all_secondary_section_lists(section)
+
+                json_data["total_course_wo_list"] =\
```

Comment:
I think this will be a lot cleaner if it's variables rather than already in the dictionary.  

secondary_lists = get_all_secondary_section_lists(section)
total_courses_wo_list +=  get_totals_course_wo_list(secondary_lists)